File: frontend/src/types/stats.ts
export interface PublicStats {
  total_items: number;
  total_categories: number;
  total_suppliers: number;
  available_items: number;
  issued_items: number;
  pending_requests: number;
  active_notices: number;
  low_stock_items: number;
  updated_at: string;
}

export interface CategoryStat {
  category_id: string;
  category_name: string;
  item_count: number;
  total_quantity: number;
}

export interface DashboardStats {
  public: PublicStats;
  categories: CategoryStat[];
  issued_by_type: Record<string, number>;
  updated_at: string;
}

File: backend/internal/models/stats.go
package models

import (
	"time"
)

type PublicStats struct {
	TotalItems      int       `json:"total_items"`
	TotalCategories int       `json:"total_categories"`
	TotalSuppliers  int       `json:"total_suppliers"`
	AvailableItems  int       `json:"available_items"`
	IssuedItems     int       `json:"issued_items"`
	PendingRequests int       `json:"pending_requests"`
	ActiveNotices   int       `json:"active_notices"`
	LowStockItems   int       `json:"low_stock_items"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type CategoryStat struct {
	CategoryID   string `json:"category_id"`
	CategoryName string `json:"category_name"`
	ItemCount    int    `json:"item_count"`
	TotalQty     int    `json:"total_quantity"`
}

type DashboardStats struct {
	PublicStats  PublicStats    `json:"public"`
	Categories   []CategoryStat `json:"categories,omitempty"`
	IssuedByType map[string]int `json:"issued_by_type,omitempty"`
	UpdatedAt    time.Time      `json:"updated_at"`
}
